apps/react-example-nested-stores/tsconfig.app.json

```diff
@@ -5,5 +5,9 @@
     "types": ["node"]
   },
   "exclude": ["**/*.spec.ts", "**/*.spec.tsx"],
-  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
+  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"],
+  "files": [
+    "../../node_modules/@nrwl/react/typings/cssmodule.d.ts",
+    "../../node_modules/@nrwl/react/typings/image.d.ts"
```

Comment:
What were these needed for specifically?

Reply:
this was ran by `@nrwl/nx` when I was instructed to run `yarn nx migrate --run-migrations=migrations.json`

Reply:
Seems like [potentially] having to maintain these configs merits figuring out the context surrounding why they're adding this (vs putting in core). Presumably its because there's some tradeoff or reason we may want to do something different? Seems odd.

Reply:
TL;DR its a TS 3.9 pattern:


Solution-style TypeScript configuration files
One significant change for Nx workspaces for is the addition of solution-style tsconfigs. Solution-style tsconfig support was added with TypeScript 3.9. Solution-style tsconfigs are a new feature of TypeScript that allows the IDE to use specific tsconfig files for references to different files. Previously, using global methods available for unit tests in application code would not be identified as an error in the IDE because all files in a project use the tsconfig.json. Now, the tsconfig.json file is a solution-style tsconfig that does not apply to any files itself but has references to the tsconfig.app.json and tsconfig.spec.json, allowing the IDE to properly show an error if types for tests are used in application code.
As part of this change, we are renaming the root tsconfig.json more appropriately to tsconfig.base.json as it should only contain the base config and is not a tsconfig file that should be directly used by the IDE.